File: Data Structures & Algorithms/maximum-subarray/submission-0.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maxSum = nums[0];//best sum seen so far;
        int currSum = nums[0];
        for(int i = 1; i < nums.size(); i++){
            int x = nums[i];
            currSum = max(x, x + currSum);
            maxSum = max(maxSum, currSum);
        }
        return maxSum;
    }
};

File: Data Structures & Algorithms/jump-game-ii/submission-0.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        jump = 0
        farthest = 0
        currEnd = 0
        for i in range(len(nums)-1):
            farthest = max(farthest, i+nums[i])
            if i == currEnd:
                jump += 1
                currEnd = farthest
        return jump
        